### DISCUSSION TITLE
helm supports the installation of all components at once

### DISCUSSION THREAD

**Discussion by nicole-lihui** [istio/istio#52595] [Ideas]
helm supports the installation of all components at once

use parent & subcharts model to install istio all compoents at once 
https://helm.sh/docs/chart_template_guide/subcharts_and_globals/

**Comment by ilrudie** [comment]
cc @bleggett who has been advocating for this type of UX for a little while now

**Comment by bleggett** [comment]
This already exists for ambient here: https://github.com/istio/istio/tree/master/manifests/sample-charts/ambient but we have decided to not document or recommend its use ATM.

I think we definitely would like to at least get to a place where we have

`helm install istio-dataplane`
`helm install istio-controlplane` 

And from there composing those two into a single Helm chart becomes ~ 3 lines of YAML - for us or anyone else who wants to do that.

But to get there there are a few things we have to do.

1. Our charts need to be composable in all cases with no bugs, which means fixing our templating and value scoping bugs/opinions. These have been burned down to some degree, but we still have a few gotchas. I added the `area/charts` issue tag to help track this stuff: https://github.com/istio/istio/issues?q=is%3Aissue+is%3Aopen+helm+label%3Aarea%2Fcharts

2. Istio needs to actually ship and test and exercise our base charts in some sort of default composition, so we fix issues in (1) and prevent ourselves from injecting more issues like that.

3. We need to decide how `istioctl` interacts with Helm, since it is kinda-but-not-exactly like Helm. I think @howardjohn and @costinm and myself are pretty aligned on "`istioctl install` should effectively just shell out to `helm install` and emit valid Helm installs you can just pick up and use with Helm tooling outside of istioctl".
   - That way `istioctl` can be simple, use Helm under the hood, continue to remain the "single binary experience", and it ensures we are always exercising our charts exactly the way plain Helm would in all cases, which is valuable for enforcing consistency. In that approach you can `istioctl install` and `helm uninstall`, or you can `helm install` and `istioctl uninstall`, or whatever permutations you like, and the entrypoint in all cases is as-close-to-identical as we can make it.

**Reply by anumalasri** [reply]
Found this discussion lately, as I am trying for the similar approach to install **Istio** using minimal helm charts.  Among Control plane components, current helm install instructions suggest to `wait` for some of the components to complete .   

May we know, Does Ambient chart work fine without following order in installing these components ?

**Reply by howardjohn** [reply]
Istio can be installed in any order, just some components depend on others so will not become Ready until they are around. But in theory you can install all at once
